src/frontend/posts_page/components/VisibilitySetting.tsx

```diff
@@ -1,14 +1,13 @@
 import React from 'react';
 
-interface PublicProps {
-}
-interface PublicState {
+interface IVisibilitySettingState {
   public: boolean,
   friendsOnly: boolean,
   private: boolean
 }
-class SetPublic extends React.Component<PublicProps, PublicState> {
-  constructor(props: PublicProps) {
+
+class VisibilitySetting extends React.Component<{}, IVisibilitySettingState> {
```

Comment:
Reason for renaming: Components shouldn't have verb as a class name. `SetPublic` is more of a action name. Also, this component sets the posts to either private, public, or toFriends, so including only `Public` in the class name is a little misleading.